pkg/controller/atlasdatafederation/connectionsecrets.go

```diff
@@ -0,0 +1,104 @@
+package atlasdatafederation
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/connectionsecret"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/stringutil"
+)
+
+func (r *AtlasDataFederationReconciler) ensureConnectionSecrets(ctx *workflow.Context, project *mdbv1.AtlasProject, df *mdbv1.AtlasDataFederation) workflow.Result {
```

Comment:
This is basically the same as https://github.com/mongodb/mongodb-atlas-kubernetes/blob/2ce7fee74e8426091f6b6aa8e2f89730af2e5dbc/pkg/controller/atlasdeployment/advanced_deployment.go#L397

What do you think about abstracting it, passing the connectionStrings/hostnames, moving it to `connectionsecret` package, and reuse in both places?

Reply:
They are not the same. Yes, this can be extracted and reused, but the method would include a lot more parameters, like event recorder, k8s client, the actual object, etc.